test/cobalt/experiments/test_rerank_quality.py

```
@@ -0,0 +1,226 @@
+        'documents': [
+            {'id': 'doc1', 'text': 'Semantic search uses embeddings to find documents based on meaning rather than exact keyword matches.', 'true_relevance': 0.95},
+            {'id': 'doc2', 'text': 'Cooking recipes for Italian pasta dishes require fresh ingredients and proper technique.', 'true_relevance': 0.0},
+            {'id': 'doc3', 'text': 'To implement semantic search, generate embeddings for your documents and store them in a vector database.', 'true_relevance': 0.90},
+            {'id': 'doc4', 'text': 'Full-text search engines like Elasticsearch use inverted indexes for fast keyword lookup.', 'true_relevance': 0.40},
+            {'id': 'doc5', 'text': 'Query embedding is generated at search time and compared against document embeddings using cosine similarity.', 'true_relevance': 0.85},
+        ],
+    },
+]
+
+
+def _simulate_rerank(query: str, documents: list[dict]) -> list[dict]:
```

Ruff (0.15.7)

[warning] Unused function argument: `query`

(ARG001)